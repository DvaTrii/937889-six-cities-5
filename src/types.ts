export type OfferCardType = {
  id: number,
  city: {
    name: string,
    coordinates: {
      latitude: number,
      longitude: number,
      zoom: number
    }
  },
  coordinates: {
    latitude: number,
    longitude: number,
    zoom: number
  },
  pictures: Array<string>,
  previewImage: string,
  isPremium?: boolean,
  isBookmark?: boolean,
  price: number,
  title: string,
  type: string,
  rating: number,
  description: string,
  bedroomsMax: number,
  guestsMax: number,
  amenities: Array<string> | null,
  hostInfo: {
    avatar: string,
    name: string,
    isSuper?: boolean,
  }
};

export enum Cities {
  PARIS = `Paris`,
  COLOGNE = `Cologne`,
  BRUSSELS = `Brussels`,
  AMSTERDAM = `Amsterdam`,
  HAMBURG = `Hamburg`,
  DUSSELDORF = `Dusseldorf`,
}

export type ReviewType = {
  id: number,
  date: number,
  text: string,
  rating: number,
  userInfo: {
    avatar: string,
    name: string,
    id: number,
    isSuper: boolean,
  }
};
