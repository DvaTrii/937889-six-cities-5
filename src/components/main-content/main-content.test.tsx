import * as React from "react";
import * as renderer from "react-test-renderer";
import {BrowserRouter} from "react-router-dom";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import MainContent from "./main-content";
import {OfferCardType} from "../../types";

const mockStore = configureStore([]);

const offers: Array<OfferCardType> = [
  {
    id: 101,
    city: {
      name: `Amsterdam`,
      coordinates: {
        latitude: 52.38333,
        longitude: 4.9,
        zoom: 12
      }
    },
    coordinates: {
      latitude: 52.3909553943508,
      longitude: 4.85309666406198,
      zoom: 12
    },
    previewImage: `img/apartment-02.jpg`,
    pictures: [`img/apartment-02.jpg`, `img/apartment-02.jpg`, `img/apartment-02.jpg`, `img/apartment-02.jpg`, `img/apartment-02.jpg`, `img/apartment-02.jpg`],
    isPremium: true,
    isBookmark: true,
    price: 100,
    title: `Wood and stone place`,
    type: `House`,
    rating: 5,
    description: `Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium`,
    bedroomsMax: 4,
    guestsMax: 5,
    amenities: [`Wi-Fi`, `Heating`, `Kitchen`, `Parking`, `Flowers`, `Dishwasher`, `Towels`, `TV`],
    hostInfo: {
      avatar: `img/avatar-max.jpg`,
      name: `Ross`,
      isSuper: true,
    },
  },
  {
    id: 102,
    city: {
      name: `Amsterdam`,
      coordinates: {
        latitude: 52.38333,
        longitude: 4.9,
        zoom: 12
      }
    },
    coordinates: {
      latitude: 52.369553943508,
      longitude: 4.85309666406198,
      zoom: 12
    },
    previewImage: `img/apartment-02.jpg`,
    pictures: [`img/room.jpg`, `img/room.jpg`, `img/room.jpg`, `img/room.jpg`, `img/room.jpg`, `img/room.jpg`],
    isPremium: false,
    isBookmark: true,
    price: 80,
    title: `Nice, cozy, warm big bed apartment`,
    type: `Hotel`,
    rating: 4,
    description: `Nemo enim ipsam voluptatem quia voluptas sit aspernatur aut odit aut fugit`,
    bedroomsMax: 4,
    guestsMax: 5,
    amenities: [`Wi-Fi`, `Heating`, `Kitchen`, `Fridge`, `Washing machine`, `Dishwasher`, `Towels`, `TV`],
    hostInfo: {
      avatar: `img/avatar-max.jpg`,
      name: `Chandler`,
      isSuper: true,
    },
  },
  {
    id: 103,
    city: {
      name: `Amsterdam`,
      coordinates: {
        latitude: 52.38333,
        longitude: 4.9,
        zoom: 12
      }
    },
    coordinates: {
      latitude: 52.3909553943508,
      longitude: 4.929309666406198,
      zoom: 12
    },
    previewImage: `img/apartment-02.jpg`,
    pictures: [`img/apartment-01.jpg`, `img/apartment-01.jpg`, `img/apartment-01.jpg`, `img/apartment-01.jpg`, `img/apartment-01.jpg`, `img/apartment-01.jpg`],
    isPremium: false,
    isBookmark: false,
    price: 120,
    title: `Canal View Prinsengracht`,
    type: `Hotel`,
    rating: 3,
    description: `Ut enim ad minima veniam, quis nostrum exercitationem ullam corporis suscipit laboriosam`,
    bedroomsMax: 1,
    guestsMax: 2,
    amenities: [`Wi-Fi`, `Heating`, `Kitchen`, `Fridge`, `Washing machine`, `Dishwasher`, `Towels`, `TV`],
    hostInfo: {
      avatar: `img/avatar-angelina.jpg`,
      name: `Monica`,
      isSuper: false,
    },
  },
];

it(`Should MainContent render properly`, () => {
  const store = mockStore({
    DATA: {
      offers,
      isLoadedOffers: true
    },
    APP: {
      city: `Amsterdam`,
      activeSorter: `Popular`,
      hoveredOfferId: -1,
    }
  });

  const tree = renderer
    .create(
        <Provider store={store}>
          <BrowserRouter>
            <MainContent />
          </BrowserRouter>
        </Provider>, {
          createNodeMock: () => document.createElement(`div`)
        }
    ).toJSON();

  expect(tree).toMatchSnapshot();
});
