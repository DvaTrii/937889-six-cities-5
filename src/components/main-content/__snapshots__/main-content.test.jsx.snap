// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should MainContent render properly 1`] = `
<div
  className="cities"
>
  <div
    className="cities__places-container container"
  >
    <section
      className="cities__places places"
    >
      <h2
        className="visually-hidden"
      >
        Places
      </h2>
      <b
        className="places__found"
      >
        3
         places to stay in 
        Amsterdam
      </b>
      <form
        action="#"
        className="places__sorting"
        method="get"
      >
        <span
          className="places__sorting-caption"
        >
          Sort by
        </span>
        <span
          className="places__sorting-type"
          onClick={[Function]}
          tabIndex="0"
        >
          Popular
          <svg
            className="places__sorting-arrow"
            height="4"
            width="7"
          >
            <use
              xlinkHref="#icon-arrow-select"
            />
          </svg>
        </span>
        <ul
          className="places__options places__options--custom
        "
        >
          <li
            className="places__option places__option--active"
            onClick={[Function]}
            tabIndex="0"
          >
            Popular
          </li>
          <li
            className="places__option "
            onClick={[Function]}
            tabIndex="0"
          >
            Price: low to high
          </li>
          <li
            className="places__option "
            onClick={[Function]}
            tabIndex="0"
          >
            Price: high to low
          </li>
          <li
            className="places__option "
            onClick={[Function]}
            tabIndex="0"
          >
            Top rated first
          </li>
        </ul>
      </form>
      <div
        className="cities__places-list places__list tabs__content"
      >
        <article
          className="cities__place-card place-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="place-card__mark"
          >
            <span>
              Premium
            </span>
          </div>
          <div
            className="cities__image-wrapper place-card__image-wrapper"
          >
            <a
              href="#"
            >
              <img
                alt="Place image"
                className="place-card__image"
                height="200"
                src="img/apartment-02.jpg"
                width="260"
              />
            </a>
          </div>
          <div
            className="place-card__info"
          >
            <div
              className="place-card__price-wrapper"
            >
              <div
                className="place-card__price"
              >
                <b
                  className="place-card__price-value"
                >
                  €
                  100
                </b>
                <span
                  className="place-card__price-text"
                >
                  / night
                </span>
              </div>
              <button
                className="place-card__bookmark-button
           place-card__bookmark-button--active
           button"
                type="button"
              >
                <svg
                  className="place-card__bookmark-icon"
                  height="19"
                  width="18"
                >
                  <use
                    xlinkHref="#icon-bookmark"
                  />
                </svg>
                <span
                  className="visually-hidden"
                >
                  In bookmarks
                </span>
              </button>
            </div>
            <div
              className="place-card__rating rating"
            >
              <div
                className="place-card__stars rating__stars"
              >
                <span
                  style={
                    Object {
                      "width": "100%",
                    }
                  }
                />
                <span
                  className="visually-hidden"
                >
                  Rating
                </span>
              </div>
            </div>
            <h2
              className="place-card__name"
            >
              <a
                href="/offer/101"
                onClick={[Function]}
              >
                Wood and stone place
              </a>
            </h2>
            <p
              className="place-card__type"
            >
              House
            </p>
          </div>
        </article>
        <article
          className="cities__place-card place-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="cities__image-wrapper place-card__image-wrapper"
          >
            <a
              href="#"
            >
              <img
                alt="Place image"
                className="place-card__image"
                height="200"
                src="img/apartment-02.jpg"
                width="260"
              />
            </a>
          </div>
          <div
            className="place-card__info"
          >
            <div
              className="place-card__price-wrapper"
            >
              <div
                className="place-card__price"
              >
                <b
                  className="place-card__price-value"
                >
                  €
                  80
                </b>
                <span
                  className="place-card__price-text"
                >
                  / night
                </span>
              </div>
              <button
                className="place-card__bookmark-button
           place-card__bookmark-button--active
           button"
                type="button"
              >
                <svg
                  className="place-card__bookmark-icon"
                  height="19"
                  width="18"
                >
                  <use
                    xlinkHref="#icon-bookmark"
                  />
                </svg>
                <span
                  className="visually-hidden"
                >
                  In bookmarks
                </span>
              </button>
            </div>
            <div
              className="place-card__rating rating"
            >
              <div
                className="place-card__stars rating__stars"
              >
                <span
                  style={
                    Object {
                      "width": "80%",
                    }
                  }
                />
                <span
                  className="visually-hidden"
                >
                  Rating
                </span>
              </div>
            </div>
            <h2
              className="place-card__name"
            >
              <a
                href="/offer/102"
                onClick={[Function]}
              >
                Nice, cozy, warm big bed apartment
              </a>
            </h2>
            <p
              className="place-card__type"
            >
              Hotel
            </p>
          </div>
        </article>
        <article
          className="cities__place-card place-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="cities__image-wrapper place-card__image-wrapper"
          >
            <a
              href="#"
            >
              <img
                alt="Place image"
                className="place-card__image"
                height="200"
                src="img/apartment-02.jpg"
                width="260"
              />
            </a>
          </div>
          <div
            className="place-card__info"
          >
            <div
              className="place-card__price-wrapper"
            >
              <div
                className="place-card__price"
              >
                <b
                  className="place-card__price-value"
                >
                  €
                  120
                </b>
                <span
                  className="place-card__price-text"
                >
                  / night
                </span>
              </div>
              <button
                className="place-card__bookmark-button
           false
           button"
                type="button"
              >
                <svg
                  className="place-card__bookmark-icon"
                  height="19"
                  width="18"
                >
                  <use
                    xlinkHref="#icon-bookmark"
                  />
                </svg>
                <span
                  className="visually-hidden"
                >
                  To bookmarks
                </span>
              </button>
            </div>
            <div
              className="place-card__rating rating"
            >
              <div
                className="place-card__stars rating__stars"
              >
                <span
                  style={
                    Object {
                      "width": "60%",
                    }
                  }
                />
                <span
                  className="visually-hidden"
                >
                  Rating
                </span>
              </div>
            </div>
            <h2
              className="place-card__name"
            >
              <a
                href="/offer/103"
                onClick={[Function]}
              >
                Canal View Prinsengracht
              </a>
            </h2>
            <p
              className="place-card__type"
            >
              Hotel
            </p>
          </div>
        </article>
      </div>
    </section>
    <div
      className="cities__right-section"
    >
      <section
        className="cities__map map"
      >
        <div
          style={
            Object {
              "height": "100%",
            }
          }
        />
      </section>
    </div>
  </div>
</div>
`;
